google/cloud/logging_v2/handlers/_helpers.py

```diff
@@ -46,59 +50,82 @@ def format_stackdriver_json(record, message):
     return json.dumps(payload)
 
 
-def get_trace_id_from_flask():
-    """Get trace_id from flask request headers.
+def get_request_data_from_flask():
+    """Get trace_id and http_request data from flask request headers.
 
     Returns:
         str: TraceID in HTTP request headers.
+        dict: data about the associated http request.
     """
     if flask is None or not flask.request:
-        return None
+        return None, None
+
+    # build http_request
+    http_request = {
+        "request_method": flask.request.method,
+        "request_url": flask.request.url,
+        "request_size": str(flask.request.content_length),
+        "user_agent": flask.request.user_agent.string,
+        "remote_ip": flask.request.remote_addr,
+        "referer": flask.request.referrer,
+    }
 
+    # find trace id
+    trace_id = None
     header = flask.request.headers.get(_FLASK_TRACE_HEADER)
+    if header:
+        trace_id = header.split("/", 1)[0]
 
-    if header is None:
-        return None
-
-    trace_id = header.split("/", 1)[0]
-
-    return trace_id
+    return http_request, trace_id
 
 
-def get_trace_id_from_django():
-    """Get trace_id from django request headers.
+def get_request_data_from_django():
+    """Get trace_id and http_request data from django request headers.
 
     Returns:
         str: TraceID in HTTP request headers.
+        dict: data about the associated http request.
     """
     request = _get_django_request()
 
     if request is None:
-        return None
+        return None, None
+
+    # build http_request
+    http_request = {
+        "request_method": request.method,
+        "request_url": request.build_absolute_uri(),
+        "request_size": request.META.get(_DJANGO_LENGTH_HEADER),
+        "user_agent": request.META.get(_DJANGO_USERAGENT_HEADER),
+        "remote_ip": request.META.get(_DJANGO_REMOTE_ADDR_HEADER),
+        "referer": request.META.get(_DJANGO_REFERER_HEADER),
```

Comment:
can some of these not be set? What happens if there is no value for a key in request.META?

Reply:
I'm believe they default to None, but good point, I'll add a test to be sure

Reply:
`{}.get("nonesuch-key")` does indeed return `None`.